test/e2e_node/image_pull_test.go

```diff
@@ -0,0 +1,258 @@
+					if event.InvolvedObject.Name == testpod.Name {
+						httpdPulled, err = getDurationsFromPulledEventMsg(event.Message)
+						framework.ExpectNoError(err)
+					} else if event.InvolvedObject.Name == testpod2.Name {
+						httpdNewPulled, err = getDurationsFromPulledEventMsg(event.Message)
+						framework.ExpectNoError(err)
+					}
+				}
+			}
+
+			// as this is serialize image pulling, the waiting duration should be almost double the duration with the pulled duration.
+			// use 1.5 as a common ratio to avoid some overlap during pod creation
+			if float32(httpdNewPulled.pulledIncludeWaitingDuration/time.Millisecond)/float32(httpdNewPulled.pulledDuration/time.Millisecond) < 1.5 &&
+				float32(httpdPulled.pulledIncludeWaitingDuration/time.Millisecond)/float32(httpdPulled.pulledDuration/time.Millisecond) < 1.5 {
+				framework.Failf("At least, one of the pull duration including waiting %v/%v should be similar with the pulled duration %v/%v",
+					httpdNewPulled.pulledIncludeWaitingDuration, httpdPulled.pulledIncludeWaitingDuration, httpdNewPulled.pulledDuration, httpdPulled.pulledDuration)
+			}
+
+		})
+
+	})
+})
+
+func prepareAndCleanup(ctx context.Context, f *framework.Framework) (testpod *v1.Pod, testpod2 *v1.Pod) {
+	httpdImage := imageutils.GetE2EImage(imageutils.Httpd)
+	httpdNewImage := imageutils.GetE2EImage(imageutils.HttpdNew)
+	testpod = &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "httpd",
+			Namespace: f.Namespace.Name,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{{
+				Name:            "httpd",
+				Image:           httpdImage,
+				ImagePullPolicy: v1.PullAlways,
+			}},
+			RestartPolicy: v1.RestartPolicyNever,
+		},
+	}
+	testpod2 = &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "httpd2",
+			Namespace: f.Namespace.Name,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{{
+				Name:            "httpd-new",
+				Image:           httpdNewImage,
+				ImagePullPolicy: v1.PullAlways,
+			}},
+			RestartPolicy: v1.RestartPolicyNever,
+		},
+	}
+
+	ginkgo.By("cleanup images")
+	_ = RemoveImage(ctx, testpod.Spec.Containers[0].Image)
+	_ = RemoveImage(ctx, testpod2.Spec.Containers[0].Image)
+	return
+}
+
+type pulledStruct struct {
+	pulledDuration               time.Duration
+	pulledIncludeWaitingDuration time.Duration
+}
+
+// getDurationsFromPulledEventMsg will parse two durations in the pulled message
+// Example msg: `Successfully pulled image \"busybox:1.28\" in 39.356s (49.356s including waiting)`
+func getDurationsFromPulledEventMsg(msg string) (pulled pulledStruct, err error) {
+	splits := strings.Split(msg, " ")
+	if len(splits) == 9 {
+		pulled.pulledDuration, err = time.ParseDuration(splits[5])
+		if err != nil {
+			return
+		}
+		// to skip '('
+		pulled.pulledIncludeWaitingDuration, err = time.ParseDuration(splits[6][1:])
+		if err != nil {
+			return
+		}
+	} else {
+		err = errors.Errorf("pull event message should be spilted to 8: %d", len(splits))
+	}
+	return
```

Comment:
```suggestion
	if len(splits) != 9 {
	        return errors.Errorf("pull event message should be spilted to 8: %d", len(splits))
	}
	pulled.pulledDuration, err = time.ParseDuration(splits[5])
	if err != nil {
		return err
	}
	// to skip '('
	pulled.pulledIncludeWaitingDuration, err = time.ParseDuration(splits[6][1:])
	if err != nil {
		return err
	}
	return nil
```

Reply:
Btw, why error message says it should be splitted to 8, but we expect `len(splits) == 9`?